File: spikes/key_bridge/ios/localhold_key_bridge/Sources/localhold_key_bridge/LocalholdKeyBridgePlugin.swift
// SPDX-License-Identifier: MPL-2.0

import Flutter
import Foundation

private let LOCALHOLD_STAGE2_SPIKE_DO_NOT_SHIP = true

/// Fail-closed Stage 2 adapter with no cryptography or key state.
public final class LocalholdKeyBridgePlugin: NSObject, FlutterPlugin, KeyBridgeHostApi {
  public static func register(with registrar: FlutterPluginRegistrar) {
    precondition(LOCALHOLD_STAGE2_SPIKE_DO_NOT_SHIP)
    let instance = LocalholdKeyBridgePlugin()
    KeyBridgeHostApiSetup.setUp(binaryMessenger: registrar.messenger(), api: instance)
  }

  func createVaultKey(request: CreateVaultKeyRequest) throws -> VaultSessionReply {
    VaultSessionReply(error: .notImplemented)
  }

  func openVaultSession(request: OpenVaultSessionRequest) throws -> VaultSessionReply {
    VaultSessionReply(error: .notImplemented)
  }

  func encryptPayload(request: EncryptPayloadRequest) throws -> PayloadReply {
    PayloadReply(error: .notImplemented)
  }

  func decryptPayload(request: DecryptPayloadRequest) throws -> PayloadReply {
    PayloadReply(error: .notImplemented)
  }

  func rewrapVaultKey(request: RewrapVaultKeyRequest) throws -> PayloadReply {
    PayloadReply(error: .notImplemented)
  }

  func closeSession(sessionHandle: String) throws -> StatusReply {
    StatusReply(error: .notImplemented)
  }

  func closeAllSessions() throws -> StatusReply {
    StatusReply(error: .notImplemented)
  }
}

File: spikes/key_bridge/android/src/main/kotlin/dev/localhold/localhold_key_bridge/LocalholdKeyBridgePlugin.kt
// SPDX-License-Identifier: MPL-2.0

package dev.localhold.localhold_key_bridge

import io.flutter.embedding.engine.plugins.FlutterPlugin

private const val LOCALHOLD_STAGE2_SPIKE_DO_NOT_SHIP = true

/** Fail-closed Stage 2 adapter with no cryptography or key state. */
class LocalholdKeyBridgePlugin : FlutterPlugin, KeyBridgeHostApi {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        check(LOCALHOLD_STAGE2_SPIKE_DO_NOT_SHIP)
        KeyBridgeHostApi.setUp(binding.binaryMessenger, this)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        KeyBridgeHostApi.setUp(binding.binaryMessenger, null)
    }

    override fun createVaultKey(request: CreateVaultKeyRequest): VaultSessionReply =
        VaultSessionReply(error = KeyBridgeErrorCode.NOT_IMPLEMENTED)

    override fun openVaultSession(request: OpenVaultSessionRequest): VaultSessionReply =
        VaultSessionReply(error = KeyBridgeErrorCode.NOT_IMPLEMENTED)

    override fun encryptPayload(request: EncryptPayloadRequest): PayloadReply =
        PayloadReply(error = KeyBridgeErrorCode.NOT_IMPLEMENTED)

    override fun decryptPayload(request: DecryptPayloadRequest): PayloadReply =
        PayloadReply(error = KeyBridgeErrorCode.NOT_IMPLEMENTED)

    override fun rewrapVaultKey(request: RewrapVaultKeyRequest): PayloadReply =
        PayloadReply(error = KeyBridgeErrorCode.NOT_IMPLEMENTED)

    override fun closeSession(sessionHandle: String): StatusReply =
        StatusReply(error = KeyBridgeErrorCode.NOT_IMPLEMENTED)

    override fun closeAllSessions(): StatusReply =
        StatusReply(error = KeyBridgeErrorCode.NOT_IMPLEMENTED)
}
